sphinx_ansible_theme/static/css/ansible.css

```diff
@@ -191,16 +191,16 @@ table {
     display: block;
     max-width: 100%;
 }
-.documentation-table td,
-.documentation-table th {
-    padding: 4px;
-    border-left: 1px solid #000;
-    border-top: 1px solid #000;
-}
+
 .documentation-table {
-    border-right: 1px solid #000;
-    border-bottom: 1px solid #000;
+    border: 0px;
 }
+
+.rst-content table,td,tr,th { border:0; }
+.rst-content th { background-color: #6AB0DE; }
```

Comment:
The color definitions should be part of the theme. I missed them since you only mentioned "borderless" in the PR's title :)

As noted in https://github.com/ansible-community/antsibull/pull/296, this applies to other tables as well, especially since you are using very generic selectors here.